### PR TITLE
Refractor

### PR DESCRIPTION
**Pull request by sagieramos**
- YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- We Follow SOLID principles.

### CODE REVIEW

File: modules/app.rb [hunk 1]
```diff
@@ -1,148 +1,38 @@
...
-      end
-      process_choice(choice)
-    rescue Interrupt
-      puts EXIT_MESSAGE
-      break
+      puts 'Invalid option. Please try again.'
     end
+  rescue Interrupt
+    puts EXIT_MESSAGE
+    break
```
Comment (Davidosky007) [inline modules/app.rb]: * When i try to run your app i get an error from this line , I was able to fix this error by replacing the break word with return. kindly look into this
image

File: modules/app.rb [hunk 2]
```diff
@@ -1,148 +1,38 @@
...
 
 class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_rentals_for_person
+  }.freeze
```
Comment (Davidosky007) [inline modules/app.rb]: * According to the [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md#:~:text=Make%20sure%20that%20the%20role%20of%20main.rb%20is%20only%20to%20start%20your%20code%20and%20ask%20the%20user%20for%20the%20initial%20choice%20of%20options%20(or%20to%20quit).) you are to Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit). kindly look into this

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @sagieramos @SamuelTakyiBoateng 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well-written PR 👍 
* No linter errors 👍 
* Good looking readme 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes. The app works fine now, 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**